# Dropping Duplicate Rows

In [5]:
import pandas as pd

url = 'https://raw.githubusercontent.com/chrisalbon/simulated_datasets/master/titanic.csv'

df = pd.read_csv(url)
df.shape

(1313, 6)

In [6]:
df.drop_duplicates().shape

(1313, 6)

## Number Of Rows In The Original DataFrame: 1313 
## Number Of Rows After Deduping: 1313

### The reason is because drop_duplicates defaults to only dropping rows that matchperfectly across all columns. Under this condition, every row in our DataFrame, data frame, is actually unique. However, often we want to consider only a subset of col‐umns to check for duplicate rows. We can accomplish this using the subset parame‐ter

In [7]:
df.drop_duplicates(subset=["Sex"])

Name PClass   Age     Sex  Survived  SexCode
0         Allen, Miss Elisabeth Walton    1st  29.0  female         1        1
2  Allison, Mr Hudson Joshua Creighton    1st  30.0    male         0        0

In [8]:
df.drop_duplicates(subset=["Sex"],keep='last')

Name PClass   Age     Sex  Survived  SexCode
1307  Zabour, Miss Tamini    3rd   NaN  female         0        1
1312       Zimmerman, Leo    3rd  29.0    male         0        0

### A related method is duplicated, which returns a boolean series denoting if a row is a duplicate or not. This is a good option if you don’t want to simply drop duplicates.